src/kernel/mconfig_dot.ml

```diff
@@ -100,17 +100,26 @@ let prepend_config ~dir:cwd (directives : directive list) config =
 
 module Configurator = struct
   type t =
+    | Dot_merlin_custom
     | Dot_merlin
     | Dune
 
   let of_string_opt = function
+    | ".merlin.custom" ->
+      Some Dot_merlin_custom
     | ".merlin" ->
       Some Dot_merlin
     | "dune-project" | "dune-workspace" ->
       Some Dune
     | _ -> None
 
+  let dot_merlin_custom =
```

Comment:
I think this part needs improvement. I'm not sure what's the best way:
- fix the binary name to `dot-merlin-custom-reader` and allow overriding the path where it is located
- allow overriding the binary name, but do stuff here to correctly pass argv[0]

Reply:
I decided on using the full path, and correctly extracting argv[0] from it (see line 139)